tools/chapel-py/src/python-types.h

```diff
@@ -57,6 +56,33 @@ struct PythonReturnTypeInfo {};
   \
   }
 
+#define DEFINE_RETURN_TYPE(TYPE, TYPESTR, WRAP, UNWRAP) \
+  template <> \
+  T_DEFINE_RETURN_TYPE(TYPE, TYPESTR, WRAP, UNWRAP)
+
+template <typename T>
+PyObject* wrapVector(ContextObject* CONTEXT, const std::vector<T>& vec) {
+  PyObject* toReturn = PyList_New(vec.size());
+  for (size_t i = 0; i < vec.size(); i++) {
+    PyList_SetItem(toReturn, i, PythonReturnTypeInfo<T>::wrap(CONTEXT, vec[i]));
+  }
+  return toReturn;
+}
+
+template <typename T>
+std::vector<T> unwrapVector(ContextObject* CONTEXT, PyObject* vec) {
+  std::vector<T> toReturn;
```

Comment:
you should declare the vector here with a reserved size, since its already known. `std::vector<T> toReturn(PyList_Size(vec));`